docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+following illustrates basic usage:
+
+```
+sources = [
+	Path('/opt/models/sushi.safetensors'),   # a local safetensors file
+	Path('/opt/models/sushi_diffusers/'),    # a local diffusers folder
+	'runwayml/stable-diffusion-v1-5',        # a repo_id
+    'runwayml/stable-diffusion-v1-5:vae',    # a subfolder within a repo_id
+	'https://civitai.com/api/download/models/63006', # a civitai direct download link
+	'https://civitai.com/models/8765?modelVersionId=10638',  # civitai model page
+	'https://s3.amazon.com/fjacks/sd-3.safetensors', # arbitrary URL
+]
+
+for source in sources:
+   install_job = installer.install_model(source)
+   
+source2key = installer.wait_for_installs()
+for source in sources:
+    model_key = source2key[source]
+	print(f"{source} installed as {model_key}")
+```
+
+As shown here, the `install_model()` method accepts a variety of
+sources, including local safetensors files, local diffusers folders,
+HuggingFace repo_ids with and without a subfolder designation,
+Civitai model URLs and arbitrary URLs that point to checkpoint files
+(but not to folders).
+
+Each call to `install_model()` will return a `ModelInstallJob` job, a
+subclass of `DownloadJobBase`. The install job has additional
+install-specific fields described in the next section.
+
+Each install job will run in a series of background threads using
+the object's download queue. You may block until all install jobs are
+completed (or errored) by calling the `wait_for_installs()` method as
+shown in the code example. `wait_for_installs()` will return a `dict`
+that maps the requested source to the key of the installed model. In
+the case that a model fails to download or install, its value in the
+dict will be None. The actual cause of the error will be reported in
+the corresponding job's `error` field.
+
+Alternatively you may install event handlers and/or listen for events
+on the InvokeAI event bus in order to monitor the progress of the
+requested installs.
+
+The full list of arguments to `model_install()` is as follows:
+
+| **Argument**     | **Type**                     | **Default** | **Description**                           |
+|------------------|------------------------------|-------------|-------------------------------------------|
+| `source`         | Union[str, Path, AnyHttpUrl] |             | The source of the model, Path, URL or repo_id |
+| `inplace`        | bool                         | True        | Leave a local model in its current location |
+| `variant`        | str                          | None        | Desired variant, such as 'fp16' or 'onnx' (HuggingFace only) |
+| `subfolder`      | str                          | None        | Repository subfolder (HuggingFace only)   |
+| `probe_override` | Dict[str, Any]               | None        | Override all or a portion of model's probed attributes |
+| `metadata`       | ModelSourceMetadata          | None        | Provide metadata that will be added to model's config |
```

Comment:
Just intended as a convenience. I was thinking about a user who wants to add a set of tags to their models automatically, but since tags are going away....

@psychedelicious Any thoughts on this?